### ISSUE TITLE
🐞 Bug: Custom tabIndex does not work

### ISSUE THREAD

**Issue by MS-v7** [public-ui/kolibri#6869]
### Link to the code that reproduces this issue:
https://stackblitz.com/edit/vitejs-vite-cmgd82?file=src%2FApp.tsx (mit onClick)

### Can you categorise where the error occurs?
React KoliBri Components

### Which browser or operating system do you used to test KoliBri?
Latest Chrome and Edge 09/2024

### How to reproduce issue?
I have > 2 Kolibri Elements in my React component and want to change the tabbing order to not follow the visual and hierachical order of the elements but what I assign via _tabindex.

### Current vs. Expected:
When giving _tabindex information to elements the tabbing still follows the visual and DOM hierarchy rather than the assing Index Numbers

### Environment informations:
```bash {
 "Operating System": {
 "platform": "win32",
 "arch": "x64",
 "version": "10.0.17763"
 },
 "Binaries": {
 "node": "v20.13.1",
 "npm": "10.5.2",
 "pnpm": "9.10.0",
 "yarn": "N/A"
 },
 "Relevant Packages": {
 "@public-ui/components": "2.1.7",
 "@public-ui/react": "2.1.7",
 "react": "18.3.1",
 "react-dom": "18.3.1",
 "typescript": "5.6.2"
 }
 } ```

**Comment by deleonio** [comment]
- [ ] Time-Slot: 1d

**Comment by laske185** [comment]
Problem: Beispielsweise der Button bietet _tabIndex an. Die Einstellung greift allerdings nicht (siehe Stackblitz-Beispiel), oder nur innerhalb des Shadow-DOMs. Damit greift aber die neu gesetzte Tab-Reihenfolge nicht. 

Mit `tabIndex` kann der TabIndex gesetzt werden, aber der Fokus wird nicht richtig gesetzt.

**Comment by Makko74** [comment]
Grundsätzlich gibt es das attr 'shadowrootdelegatesfocus' dafür.

Beispiel:

```html
<body>
    <button tabindex="2">Test</button>
    <button tabindex="5">Test 2</button>
    <button tabindex="4">Test 3</button>
    <my-button tabindex="3">
        <template shadowrootmode="open" shadowrootdelegatesfocus>
            <button onclick="alert('test')" >shadow button</button>
        </template>
    </my-button>
</body>
```

**Comment by Makko74** [comment]
Unter stencil muss es mit 'delegatesFocus' wie folgt angegeben werden:

```html
@Component({
	tag: 'kol-button',
	styleUrls: {
		default: './style.scss',
	},
	shadow: {
		delegatesFocus: true,
	},
})
```

**Comment by AlexanderSchmidtCE** [comment]
Ich hab erstmal nur die Komponente erweitert, die ein tabIndex als prop haben kann. Dabei funktioniert nicht jeder Komponente problemlos. Es kann dazu führen, dass Nebeneffekte verursacht werden, wenn auf den Container geklickt wird, auch wenn in die leere Fläche des Containers geklickt wird z.B.

KolAlert - Der Tooltip des Schließen-Buttons wird angezeigt.
KolTabs - Tabs werde fokusiert
KolToolbar - Toolbar wird fokussiert und Tooltip des ersten Buttons wird gezeigt.

Diese Komponente habe ich nciht erweitert. Bei den restlichen gab es keine zusätzlichen Nebeneffekte.

**Comment by AlexanderSchmidtCE** [comment]
Die Änderungen bewirken grundsätzlich, dass die jeweilige Komponente fokussiert wird, wenn Label oder ErrorMessage angeklickt werden.
Folgend ist eine Auflistung alles Betroffener Komponente und wie das Verhalten vor der Änderung war:

Klicks auf Label oder ErrorMessage fokussieren die Komponente:
- KolInputColor
-  KolInputEmail,
- KolInputFile
-  KolInputPassword
-  KolInputText
-  KolCombobox,
- KolSelect
-  KolTextarea

Klicks auf Label fokussieren die Komponente. ErrorMessage hat keine Auswirkung:
- KolInputDate
- KolInputNumber
- KolInputRange

Klicks auf Label und ErrorMessage verursachen kein Fokus:
- KolInputRadio
- KolSingleSelect

Bei KolInputRadio führt die Änderung zusätzlich dazu, dass die Komponente fokussiert wird, wenn eine Option mit disabled angeklickt wird.

**Comment by deleonio** [comment]
@AntnSaj, @ElVariablo: Bitte testet mal alle Buttons, Inputs, Select, Textarea und Links durch. Und prüft sie hinsichtlich der Barrierefreiheit.

Das Wichtige an dieser Änderung ist, dass egal wo man hinklickt, wird immer das interaktive Element fokussiert.

**Comment by cbraehmig** [comment]
Aus Sicht der Barrierefreiheit passt das Verhalten so. Ist ebenso ein Aspekt der Usability (Erwartungskonformität)
